app/models/manager_refresh/inventory/core.rb

```diff
@@ -43,7 +43,7 @@ def has_configuration_script_payloads(options = {})
 
     def has_configuration_workflows(options = {})
       has_inventory({
-        :model_class                 => ::ConfigurationWorkflow,
+        :model_class                 => ManageIQ::Providers::AutomationManager::ConfigurationWorkflow,
```

Comment:
`has_configuration_scripts` should have a similar model_class to isolate it to AutomationManager::ConfigurationScript because you want to filter only those on that half of the equation.

Reply:
Actually...this hardcoding should not be in core...I think I'm missing something fundamental about the refresh modeling to understand this.

Reply:
@Ladas bump

Reply:
let us brainstorm with @agrare and @slemrmartin on this

Reply:
Well for one we got rid of this way of defining inventory collections so this will never get used :)

Reply:
@agrare hm right, I got lost in what got merged. So the new way seems to already solve this? https://github.com/Ladas/manageiq/blob/dd1215d37906f6fd7ee757cdd5510a019bfa7039/app/models/manager_refresh/inventory/automation_manager.rb#L38 , right @slemrmartin  ?

Reply:
@jameswnl as @agrare wrote `has_inventory` interface is deprecated and will be removed. 
Please use `add_collection` and `ManagerRefresh::InventoryCollection::Builder::AutomationManager` instead. 

Model class is derived automatically, but, as I described https://github.com/ManageIQ/manageiq/pull/17574#issuecomment-405956539, autoloading (including calling `new`) is weird when you doesn't have IC object's class defined for each manager (EmbeddedAnsible etc.).

You can hardcode `:model_class` as you defined above in `ManageIQ::Providers::AnsibleTower::Inventory::Persister::AutomationManager`
And redefine it, if you need, in persister's subclasses